File: client/src/main/js/components/reports/CodeStyleReport.js
import React from 'react';

export function CodeStyleReport(props) {
    const latestCodeStyleReport = props.solution.codeStyleReports.last();

    if (latestCodeStyleReport) {
        return (
            <section className="report">
                <div className={'code-style-grade code-style-grade-' + latestCodeStyleReport.grade}/>
            </section>
        );
    } else {
        return <section className="report"/>;
    }
}

File: frontend/src/main/kotlin/org/flaxo/frontend/component/report/CodeStyleReport.kt
package org.flaxo.frontend.component.report

import org.flaxo.common.data.Solution
import react.RBuilder
import react.dom.div

fun RBuilder.codeStyleReport(solution: Solution) =
        solution.codeStyleReports
                .lastOrNull()
                ?.also { div(classes = "code-style-grade code-style-grade-" + it.grade) { } }
                ?: div { }
